File: LeetCode/Easy/Move Zeroes/Move Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        right = len(nums)-1
        left = 0
        while left < right:
            if nums[left] == 0:
                nums.append(nums[left])
                nums.pop(left)
                right -= 1
            else:
                left += 1   

File: LeetCode/Easy/Move Zeroes/Move Zeroes.go
func moveZeroes(nums []int)  {
    n := len(nums)
    left, right := 0, n-1
    res := make([]int, n)
    
    for _, val := range nums{
        if val == 0{
            res[right] = 0
            right --
        }else{
            res[left] = val
            left ++
        }
    }
    for idx, val := range res{
        nums[idx] = val
    }
}
